batcher/aligned-batcher/src/lib.rs

```diff
@@ -1167,39 +1178,83 @@ impl Batcher {
-            }
-            Err(_) => {
-                let receipt = try_create_new_task(
+        let result = retry_function(
+            || {
+                create_new_task_retryable(
                     batch_merkle_root,
-                    batch_data_pointer,
-                    proof_submitters,
-                    fee_params,
+                    batch_data_pointer.clone(),
+                    proof_submitters.clone(),
+                    fee_params.clone(),
+                    self.transaction_wait_timeout,
+                    &self.payment_service,
                     &self.payment_service_fallback,
                 )
-                .await?;
-
-                Ok(receipt)
+            },
+            DEFAULT_MIN_RETRY_DELAY,
+            DEFAULT_BACKOFF_FACTOR,
+            DEFAULT_MAX_RETRIES,
+        )
+        .await;
+        match result {
+            Ok(receipt) => Ok(receipt),
+            Err(RetryError::Permanent(BatcherError::ReceiptNotFoundError)) => {
+                self.metrics.canceled_batches.inc();
+                self.cancel_create_new_task_tx(fee_params.gas_price).await;
+                Err(BatcherError::ReceiptNotFoundError)
             }
+            Err(_) => Err(BatcherError::TransactionSendError),
         }
     }
 
+    // Sends a transaction to Ethereum with the same nonce as the previous one to override it.
+    // We use the maximum number of retries here because this transaction must be included.
+    pub async fn cancel_create_new_task_tx(&self, old_tx_gas_price: U256) {
+        info!("Cancelling createNewTask transaction...");
+        let iteration = Arc::new(Mutex::new(0));
+        let previous_gas_price = Arc::new(Mutex::new(old_tx_gas_price));
+
+        if let Err(e) = retry_function(
+            || async {
+                let mut iteration = iteration.lock().await;
+                let mut previous_gas_price = previous_gas_price.lock().await;
+
+                let current_gas_price = match get_gas_price(
+                    self.batcher_signer.provider(),
+                    self.batcher_signer_fallback.provider(),
+                )
+                .await
+                {
+                    Ok(gas_price) => gas_price,
+                    Err(e) => return Err(RetryError::Transient(e)),
+                };
+
+                let bumped_gas_price =
+                    get_bumped_gas_price(*previous_gas_price, current_gas_price, *iteration);
+
+                *iteration += 1;
+                *previous_gas_price = bumped_gas_price;
+
+                cancel_create_new_task_retryable(
+                    &self.batcher_signer,
+                    &self.batcher_signer_fallback,
+                    bumped_gas_price,
+                    self.transaction_wait_timeout,
+                )
+                .await
+            },
+            DEFAULT_MIN_RETRY_DELAY,
+            DEFAULT_BACKOFF_FACTOR,
+            usize::MAX,
```

Comment:
I would set a limited number of retries, as we might batcher get stuck here and I think that is worse than losing a transaction.

Reply:
Also, I think we should send an error message to the clients to finalize the connection.

Reply:
We use a `MAX` number of retries here because we won’t be able to create a new task until this transaction is included in Ethereum. We also won’t get stuck here as we are bumping the transaction with each attempt.

Reply:
Wouldn't that lead to unbounded gas costs?

Reply:
Is there an issue? Can you add a `TODO` comment?

Reply:
Done! Created #1380 and added a reference to the issue in a `TODO` comment.

Reply:
Solved